RecoLocalTracker/Phase2TrackerRecHits/src/VectorHitBuilderAlgorithm.cc

```diff
@@ -0,0 +1,559 @@
+#include "RecoLocalTracker/Phase2TrackerRecHits/interface/VectorHitBuilderAlgorithm.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit2D.h"
+#include "RecoLocalTracker/Phase2TrackerRecHits/interface/VectorHitBuilderAlgorithm.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "RecoLocalTracker/Records/interface/TkPhase2OTCPERecord.h"
+#include "RecoLocalTracker/Phase2TrackerRecHits/interface/Phase2StripCPE.h"
+
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+
+VectorHitBuilderAlgorithm::VectorHitBuilderAlgorithm(const edm::ParameterSet& conf) :
+  cpeTag_(conf.getParameter<edm::ESInputTag>("CPE")),
+  nMaxVHforeachStack(conf.getParameter<int>("maxVectorHitsinaStack")),
+  barrelCut(conf.getParameter< std::vector< double > >("BarrelCut")),
+  endcapCut(conf.getParameter< std::vector< double > >("EndcapCut")),
+  theFitter(new LinearFit())
```

Comment:
Make it a regular member, no need for `new`.

Reply:
Actually `LinearFit` has no data members so it does not have to be a member variable here either. Instead, just construct `LinearFit` when needed.